test/Kestrel.Core.Tests/Http2StreamTests.cs

```diff
@@ -964,12 +964,6 @@ public async Task ContentLength_Received_SingleDataFrameOverSize_Reset()
         [Fact]
         public async Task ContentLength_Received_SingleDataFrameUnderSize_Reset()
         {
-            // I hate doing this, but it avoids exceptions from MemoryPool.Dipose() in debug mode. The problem is since
-            // the stream's ProcessRequestsAsync loop is never awaited by the connection, it's not really possible to
-            // observe when all the blocks are returned. This can be removed after we implement graceful shutdown.
-            Dispose();
-            InitializeConnectionFields(new DiagnosticMemoryPool(KestrelMemoryPool.CreateSlabMemoryPool(), allowLateReturn: true));
```

Comment:
This is no longer called in any of our unit tests, right?

Reply:
There are a few spots:
https://github.com/aspnet/KestrelHttpServer/blob/release/2.2/test/Kestrel.Core.Tests/DiagnosticMemoryPoolTests.cs#L176-L200. This is specifically testing those behaviours so we'll keep these.
https://github.com/aspnet/KestrelHttpServer/blob/release/2.2/test/Kestrel.Transport.FunctionalTests/Http2/ShutdownTests.cs#L100. This is the abortive test and I think it's unavoidable.
https://github.com/aspnet/KestrelHttpServer/blob/0e99235d59031106bfe6c593757b99ff99f261f7/test/Kestrel.Transport.FunctionalTests/MaxRequestBufferSizeTests.cs#L117. I'll see if these can be removed though.

Reply:
Don't worry about the functional tests. I was only asking about unit tests.

The MaxRequestBufferSizeTests have been that way in a while. Even if it passes without allowing late returns on your machine, I would be very concerned about introducing test flakiness. You're also right about it being unavoidable for the abortive Shutdown tests.